### ISSUE TITLE
Feature request: Wayland support

### ISSUE THREAD

**Issue by MastaG** [ptitSeb/box86#436]
My platform only supports Wayland (running Gnome shell or Weston).
In order for this to work I guess the libwayland-client and libwayland-egl libraries should be wrapped and furthermore gl4es should support Wayland as well.

**Comment by ptitSeb** [comment]
I need to check what those libs do and how they interact with libGL first.

**Comment by MastaG** [comment]
Now that wine's wayland support is close to being merged, would it possible to have a look at wayland cliënt support?

**Comment by porschemad911** [comment]
I have been testing the wine wayland driver with box86 at commit c5bccdf300ff5017b7225e49a934c395c51f297b and x86 wine 9.16 from https://github.com/Kron4ek/Wine-Builds/releases/tag/9.16. Am hitting errors similar to those mentioned here for box64: https://github.com/ptitSeb/box64/issues/1581

**Comment by porschemad911** [comment]
Have confirmed with the author that the wine builds at https://github.com/Kron4ek/Wine-Builds/releases/tag/9.16 include the wayland driver. `winewayland.drv` does indeed exist at `/usr/lib/wine/i386-windows/winewayland.drv` and `~/.wine/dosdevices/c\:/windows/system32/winewayland.drv`

**Comment by ptitSeb** [comment]
box86 does not support wayland. Only box64 does.

**Comment by shoddydev** [comment]
> box86 does not support wayland. Only box64 does.

do you mean with wine? because i've got linux x86 apps to work on wayland using box86.

i really would like to get box86 to work with wine wayland though...

**Comment by ptitSeb** [comment]
> > box86 does not support wayland. Only box64 does.
> 
> do you mean with wine? because i've got linux x86 apps to work on wayland using box86.
> 
> i really would like to get box86 to work with wine wayland though...

What I mean is that none of the Wyland libs are wrapped. So to get wayland running there, you need to use x86 version of the libs, which might then trigger some more issue with some low level stuffs not being supported.

**Comment by shoddydev** [comment]
> What I mean is that none of the Wyland libs are wrapped. So to get wayland running there, you need to use x86 version of the libs, which might then trigger some more issue with some low level stuffs not being supported.

i see. i'm not really familiar with all the technical but ive been trying to get wine wayland working with box86 for the past few days. ive tried wine staging, the official wine 9.9, so are you saying it's possible if i were to retrieve the linux x86 libwayland-client.so libs for it to emulate that it might work? if so i'll give that a try. im on a playstation classic btw. ive been using Xwayland but its software only so its slow

**Comment by shoddydev** [comment]
for this linux x86 game called Teeworlds it uses sdl2 to wrap it. and i guess that's a way to use wayland on it as sdl2 handles the wayland stuff i dont know.

https://github.com/user-attachments/assets/ac0696ff-3c31-4b35-9d06-5e35763a77f5

maybe theres a way to have sdl2 render wine, that'll be awesome

**Comment by ptitSeb** [comment]
Yes, SDL2 is wrapped so Wayland there is "automatic". It will not be like that for Wine. I have added all the needed stuffs in box64 already, and there is a lot of work needed. That's why I have not done it, to much work for not enough payoff.

**Comment by shoddydev** [comment]
im not even sure how the hell i got it to work but i asked google gemini for help on wrapping it, i think its pretty much all emulated but here it is, wine-staging 9.12 on weston 6.0.0 and wayland 1.18 i think? it had me downloading libs from debian.org

https://github.com/user-attachments/assets/e937b338-6b20-48c6-b6dc-562934c85beb

maybe its possible to do it without wrapping if i get the exact libs? i dont know. also im on armv7, the psc uses linux 4.4.22 and glibc 2.24
